### PR TITLE
runtime-rs:merge runtime rs to main

### PR DESCRIPTION
**Pull request by studychao**
After several months of work, we have pushed all the fundamental code for rust runtime and optional built-in vmm dragonball.

And we have established CI for runtime-rs, which you could see the detail here: https://github.com/kata-containers/tests/pull/4986

### CODE REVIEW

**Comment by BbolroC** [comment]
error message on s390x:

```
13:08:56 make -C src/runtime-rs
13:08:56 make[1]: Entering directory '/home/jenkins/workspace/kata-containers-2.0-ubuntu-s390x-unit-PR/go/src/github.com/kata-containers/kata-containers/src/runtime-rs'
13:08:56 ../../utils.mk:153: "WARNING: s390x-unknown-linux-musl target is unavailable"
13:08:56 Makefile:25: *** "ERROR: invalid architecture: 's390x'".  Stop.
```

**Comment by studychao** [comment]
> error message on s390x:
> 
> ```
> 13:08:56 make -C src/runtime-rs
> 13:08:56 make[1]: Entering directory '/home/jenkins/workspace/kata-containers-2.0-ubuntu-s390x-unit-PR/go/src/github.com/kata-containers/kata-containers/src/runtime-rs'
> 13:08:56 ../../utils.mk:153: "WARNING: s390x-unknown-linux-musl target is unavailable"
> 13:08:56 Makefile:25: *** "ERROR: invalid architecture: 's390x'".  Stop.
> ```

Thanks, I have noticed this. Our built-in VMM currently doesnt have support for s390x but our rust runtime should works well with s390x.
We don't have support for other VMM in runtime-rs now, and we will add support for other VMM like QEMU in future stages so s390x will be supported soon with QEMU.

**Comment by BbolroC** [comment]
> > error message on s390x:
> > ```
> > 13:08:56 make -C src/runtime-rs
> > 13:08:56 make[1]: Entering directory '/home/jenkins/workspace/kata-containers-2.0-ubuntu-s390x-unit-PR/go/src/github.com/kata-containers/kata-containers/src/runtime-rs'
> > 13:08:56 ../../utils.mk:153: "WARNING: s390x-unknown-linux-musl target is unavailable"
> > 13:08:56 Makefile:25: *** "ERROR: invalid architecture: 's390x'".  Stop.
> > ```
> 
> Thanks, I have noticed this. Our built-in VMM currently doesnt have support for s390x but our rust runtime should works well with s390x. We don't have support for other VMM in runtime-rs now, and we will add support for other VMM like QEMU in future stages so s390x will be supported soon with QEMU.

Thanks for the info. ^^

**Comment by liubin** [comment]
> Install runtime
.ci/install_runtime.sh: line 49: KATACONTAINERS_REPO: unbound variable
[install_kata.sh:54] ERROR: .ci/install_runtime.sh

File: .github/workflows/move-issues-to-in-progress.yaml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
             exit 1
           }
 
-          project_name="Issue backlog"
+          project_name="runtime-rs"
```
Comment (liubin) [inline .github/workflows/move-issues-to-in-progress.yaml:62]: Should we revert this back to `Issue backlog`?
/cc @jodh-intel @kata-containers/architecture-committee

Comment (jodh-intel) [reply .github/workflows/move-issues-to-in-progress.yaml:62]: Good catch @liubin ! ;)

File: docs/install/kata-containers-3.0-rust-runtime-installation-guide.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    
+    Example for `x86_64`
+    ```
+    $ rustup target add x86_64-unknown-linux-musl
+    ```
+* [Musl `libc`](http://musl.libc.org/) install
+
+    Example for musl 1.2.3
+    ```
+    $ wget https://git.musl-libc.org/cgit/musl/snapshot/musl-1.2.3.tar.gz
```
Comment (jodh-intel) [inline docs/install/kata-containers-3.0-rust-runtime-installation-guide.md:71]: Nit: consistency: `curl` was used above and that's the defaultdownloader so I'd `s/wget/curl/`.

**Review by jodh-intel — APPROVED** [review]
Thanks @studychao - Wow, and I thought the other `runtime-rs` PRs were big! :smile:

Based on the fact that I've reviewed all the `runtime-rs` PRs individually, I'm approving this "mega rollup" PR but it's clearly too big to do a formal (re-)review on all the code here.

I'd like to say thank you very much to you and the rest of the team for all the work you've put into what is a huge and highly significant project. It's a major achievement and an extremely impressive one!! :fireworks: :fireworks: :fireworks: :champagne: :wine_glass: :balloon: :balloon: :balloon: 

___

Aside: Is there a list somewhere of outstanding work items? I'm thinking of things like:

- Additional work on supporting the existing hypervisors.
- Changes to the `tests` repo for this new runtime+hypervisor
- Any further doc updates (design, installation)?
- Addition of tracing.
- _etc_.

**Review by GabyCT — APPROVED** [review]

File: docs/install/kata-containers-3.0-rust-runtime-installation-guide.md (hunk 2 above)
Comment (studychao) [reply docs/install/kata-containers-3.0-rust-runtime-installation-guide.md:71]: @Tim-0731-Hzt Please help modify the document.

**Comment by studychao** [comment]
@jodh-intel Hi Jodh, thanks a lot for your review work and all the support :)
We are preparing for all the work you listed and you could see that in the community within few weeks.

**Pull request merged**